COMMANDS.org

```diff
@@ -0,0 +1,117 @@
+#+end_quote
+
+Search for packages via:
+- The web UI @ https://search.nixos.org/
+- Using the cli:
+  #+begin_src bash
+  nix-env -qaP 'some_package' 
+  #+end_src
+
+  - ~-q~: Stands for "query". Used to query the Nix store for /installed/ pacakges
+  - ~-a~: Stands for "avalible". Used to search for /avalible packages/ to be installed
+  - ~-P~: Includes the attribute path of the package. This result is often used with the ~-A~ flag during installs to more precisely target the desired package.
+
+** 🪛 Imperative Package Management
+
+#+begin_quote
+/Question: I'm used to tools like brew - how do I use nix in a similar way?/
+#+end_quote
+
+If your workflow has involved...
+- ~brew install some_package~
+- ~port install some_package~
+- ~api-get install some_package~
+
+...and are making the move to ~nix~, then you are in the right place! ~nix~ can be used as a drop in replacement for these tools on most linux/unix based systems.
+
+*** Install a package
+
+Once the desired package has been identified, use ~nix-env~ to install the package globally...
+
+#+begin_src bash
+  nix-env -i some_package
+#+end_src
+
+- ~-i~: short flag for ~--install~.
+  - Takes multiple arguements in a space-seperated list.
+  - By default, ~nix-env -i~ will do a lookup of the argument given against ~nixpkgs~, similarly to going to https://search.nixos.org/, searching, and using the top result.
+- ~-A~: short for ~--attr~ across many ~nix~ commands.
+  - Instead of relying on the lookup, which could be slow at times, one could add ~-A~ along with the /exact/ package name to skip the lookup completely. This requires the name given to be exact, and can allow for installing a specific version of the target package.
+  - Can be combined with other flags i.e. ~-iA~
+
+#+begin_quote
+Note: If a package being installed complains about the package license being "unfree", this means the package has some licensing agreement that is not conguent with typical open-source tools.
+
+To install these packages anyway, prepend ~NIXPKGS_ALLOW_UNFREE=1~ to the call to make a 1 time allowance. The whole call should look something like:
+#+begin_src bash
+  NIXPKGS_ALLOW_UNFREE=1 nix-env -i some_package
+#+end_src
+#+end_quote
+
+*** Uninstalling a package
+
+Uninstalling a package requires knowing its name. We can find a list of the installed packages like this:
+
+#+begin_src bash
+  nix-env -q some_package
+#+end_src
+
+With the target package name in hand, we can then run:
+
+#+begin_src bash
+  nix-env -e some_package
+#+end_src
+
+- ~-e~: Stands for "erase"
+  - Takes a space-seperated list of pacakges to be removed from the system
+
+To validate, simply re-run ~nix-env -q some_package~ and check that output.
+
+** 📝 Declaritive Pacakge Management
+
+To be detailed later.
+
+There are a few ways to approach this, including using ~flakes~ or helper packages installed imperatively like ~home-manager~
+
+** 🗑️ Temporary Packages
```

Comment:
This section is accurate...but this is sort of a new concept coming from brew-land.

Some alternative titles here:

* Ephemeral Packages
* One time use Packages
* Single use Package
* Did you know you can give a package a test run?
* Did you know you can give a package a trial run?
* Previewing a package without actually installing it
* Temporary Packages

Reply:
You want use "ephemeral" here, but declarative earlier was too much?

Reply:
Oh I hate most of these suggestions, I don't know if there's a good option

I use the word ephemeral...or at least see it in games I play lol